scripts/run-callgrind.sh

```
@@ -0,0 +1,85 @@
+    echo "Error: Binary not found: $BINARY"
+    echo ""
+    echo "Usage: $0 [binary] [args...]"
+    echo "Example: $0 bin/moac"
+    echo ""
+    echo "Tip: Build with debug symbols for better results:"
+    echo "  make clean && make linux DEBUG=-g"
+    exit 1
+fi
+
+shift 2>/dev/null || true
+ARGS="$@"
```

Shellcheck (0.11.0)

[warning] Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.

(SC2124)